kafkakewl-metrics/src/main/scala/com/mwam/kafkakewl/metrics/Main.scala

```diff
@@ -56,7 +56,7 @@ object Main extends ZIOAppDefault {
 
     (for
       topicInfoSource <- ZIO.service[KafkaTopicInfoSource]
-      _ <- HighOffsetMetricsExposer.startPublishing(topicInfoSource.subscribeStream())
+      _ <- ZIO.service[HighOffsetMetricsExposer]
```

Comment:
yeah, I've been thinking about it but couldn't figure out anything better. The thing I remembered vaguely was https://zio.dev/reference/contextual/zlayer/#converting-a-layer-to-a-zio-application but it does not help here.

Maybe move this line below 62 to logically separate from the `startPublishing()` calls and add a comment why we have this here?